packages/@lwc/integration-karma/test-hydration/mismatches/invalid-number-of-nodes/if-true/index.spec.js

```diff
@@ -14,12 +14,21 @@ export default {
         const hydratedSnapshot = this.snapshot(target);
 
         expect(hydratedSnapshot.text).not.toBe(snapshots.text);
-
-        TestUtils.expectConsoleCallsDev(consoleCalls, {
-            error: [
-                'Server rendered more nodes than the client.',
-                'Hydration completed with errors.',
-            ],
-        });
+        if (process.env.DISABLE_STATIC_CONTENT_OPTIMIZATION) {
+            TestUtils.expectConsoleCallsDev(consoleCalls, {
+                warn: [
+                    'Hydration mismatch: text values do not match, will recover from the difference',
+                    'Server rendered more nodes than the client.',
+                    'Hydration completed with errors.',
+                ],
+            });
+        } else {
+            TestUtils.expectConsoleCallsDev(consoleCalls, {
+                warn: [
+                    'Server rendered more nodes than the client.',
+                    'Hydration completed with errors.',
+                ],
+            });
+        }
```

Comment:
Why does the text mismatch with `DISABLE_STATIC_CONTENT_OPTIMIZATION` ? 🤔

Tested with all flags used in CI. This was the only issue.

Reply:
iirc it's something to do with how we concatenate adjacent text nodes in different scenarios, particularly how we use `\u200d` (zero-width joiner) as a placeholder

Reply:
In the static content optimization we only call `hydrateStaticParts`, not `hydrateElement`+`hydrateText`. So you may see slightly different hydration warnings.